test/stage1_test.rb

```diff
@@ -66,4 +72,69 @@
       expect(subject.devices).to eq([])
     end
   end
+
+  describe "#add_udev_device" do
+    it "adds underlayed disk device for lvm disk" do
+      allow(Yast::BootStorage).to receive(:underlaying_devices).and_call_original
+      Yast::BootStorage.instance_variable_set(:@underlaying_devices_cache, {})
+      target_map_stub("storage_lvm.yaml")
+
+      allow(Yast::BootStorage).to receive(:mbr_disk)
+        .and_return("/dev/system")
+      allow(Yast::BootStorage).to receive(:BootPartitionDevice)
+        .and_return("/dev/system/root")
+      allow(Yast::BootStorage).to receive(:RootPartitionDevice)
+        .and_return("/dev/system/root")
+
+      subject.add_udev_device("/dev/system")
+
+      expect(subject.devices).to eq(["/dev/vda"])
+
+      expect(subject.mbr?).to eq true
+    end
+
+    it "adds underlayed partition devices for lvm partition" do
+      allow(Yast::BootStorage).to receive(:underlaying_devices).and_call_original
+      Yast::BootStorage.instance_variable_set(:@underlaying_devices_cache, {})
+      target_map_stub("storage_lvm.yaml")
+
+      allow(Yast::BootStorage).to receive(:mbr_disk)
+        .and_return("/dev/system")
+      allow(Yast::BootStorage).to receive(:BootPartitionDevice)
+        .and_return("/dev/system/root")
+      allow(Yast::BootStorage).to receive(:RootPartitionDevice)
+        .and_return("/dev/system/root")
+
+      subject.add_udev_device("/dev/system/root")
+
+      expect(subject.devices).to eq(["/dev/vda3"])
+
+      expect(subject.boot_partition?).to eq true
+    end
+  end
+
+  describe "#can_use_boot?" do
+    before do
+      allow(subject).to receive(:can_use_boot?).and_call_original
+    end
+
+    it "returns false if boot partition fs is xfs" do
+      target_map_stub("storage_xfs.yaml")
+
+      allow(Yast::BootStorage).to receive(:mbr_disk).and_return("/dev/vda")
+      allow(Yast::BootStorage).to receive(:BootPartitionDevice).and_return("/dev/vda1")
+
+      expect(subject.can_use_boot?).to eq false
+    end
+
+    it "returns true otherwise" do
```

Comment:
in fact it was list of conditions where it returns false and this was the last one to specifi rest of states